<launch>
  <arg name="pose_x" default="16.6"/>
  <arg name="pose_y" default="17"/>
  <arg name="pose_th" default="-1.57"/>
  <arg name="map" default="profest"/>
  <node pkg="map_server" type="map_server" name="map_server" args="$(find graph_route)/maps/$(arg map).yaml"/>
  <node pkg="odometry" type="odometry_node" name="odometry_node" output="screen">
    <param name="initial_x" value="$(arg pose_x)"/>
    <param name="initial_y" value="$(arg pose_y)"/>
    <param name="initial_th" value="$(arg pose_th)"/>
    <param name="base_l"  value="0.333"/>
    <param name="wheel_d"  value="0.150"/>
    <param name="rudder_k"  value="0.96"/>
  </node>
  <node pkg="tf" type="static_transform_publisher" name="base_laser_link" args="-0.18 0.0 0.0 0.0 0.0 0.0 /base_link /base_laser_link 50" />
  <node pkg="tf" type="static_transform_publisher" name="odom_to_map" args="0 0 0.151 0 0 0 1.0 map odom 100" />
  <node pkg="graph_route" type="graph_route_node" name="graph_route_node" output="screen">
    <rosparam file="$(find graph_route)/config/graph_$(arg map).yaml" command="load" />
    <param name="max_rudder"  value="0.4"/>
    <param name="min_speed"  value="0.2"/>
    <param name="max_speed"  value="0.3"/>
  </node>
  <!--
  <node pkg="immo_planner" type="immo_planner_node" name="immo_planner_node" output="screen">
    <param name="max_rudder"  value="0.4"/>
    <param name="min_speed"  value="0.4"/>
    <param name="max_speed"  value="0.2"/>
  </node>
-->
</launch>
